py/bitbox02/bitbox02/communication/bitbox_api_protocol.py

```diff
@@ -407,3 +419,22 @@ def get_info(self) -> Tuple[str, Platform, Union[BitBox02Edition, BitBoxBaseEdit
         unlocked_byte = response[0]
         unlocked = {0x00: False, 0x01: True}[unlocked_byte]
         return (version_str, platform, edition, unlocked)
+
+    def check_version(self) -> Tuple[bool, semver.VersionInfo]:
```

Comment:
This should check for minimum supported and minimum not supported, otherwise the library will break with each firmware upgrade, even though most firmware upgrades would be compatible.

I'd do it the same ways as in the app:

https://github.com/digitalbitbox/bitbox02-api-go/blob/master/api/firmware/device.go#L32-L37

It can happen e.g. that the range of supported versions is 6.3.0 to 8.0.0.

Reply:
As discussed I limit the supported range within the lowest supported in the next highest major version. Meaning 4.9.9 is supported if the lowest supported is 4.8.0, but 5.0.0 is no longer supported. Both upper and lower version mismatches now throw their own exception, each with their own unique text.